### DISCUSSION TITLE
NETCDF_NCCF for MODE-TD

### DISCUSSION THREAD

**Discussion by lswenson** [dtcenter/METplus#2569] [File I/O] [answered]
Hello,
I've been running MODE successfully using some not quite cf compliant netcdf files. I'd like to now run MTD on these same files but I can't find an equivalent to "MODE_FCST_FILE_TYPE = NETCDF_NCCF" for MTD. Do I need to find a way to make these files cf compliant or is there a solution you could help me find? Let me know if you'd like an example file or anything else and I can get that uploaded. Thank you!

**Comment by jprestop** [comment]
Hi @lswenson.  Thank you for your question.  As you indicated, a MODE_FCST_FILE_TYPE options exists, but no such MTD_FCST_FILE_TYPE exists.  @georgemccabe, could you please reply with the best course of action for this user?

@lswenson, @georgemccabe is out and will likely not be able to reply today.

**Comment by georgemccabe** [comment, marked as answer]
Hi @lswenson, thank you for bringing to our attention that MTD config options to set the fcst and obs file_type value do not exist. I created issue dtcenter/METplus#2570 to track the work to add those missing config options.

In the meantime, there are 2 ways to specify that your input files are "not quite CF compliant":

1) You should be able to set the `file_type` for each field separately using the `FCST_VAR<n>_OPTIONS` and `OBS_VAR<n>_OPTIONS` config variables.

```
FCST_VAR1_OPTIONS = file_type = NETCDF_NCCF;
OBS_VAR1_OPTIONS = file_type = NETCDF_NCCF;
```

You will have to do that for each field that you have defined in your config file.

2) Another option is to set the file type using `MTD_MET_CONFIG_OVERRIDES` (see https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html#overriding-unsupported-met-configuration-variables for more information).

```
MTD_MET_CONFIG_OVERRIDES = fcst = { file_type = NETCDF_NCCF; } obs = { file_type = NETCDF_NCCF; }
```

If you go with the latter approach, you will be able to replace the MET config overrides setting with `MTD_FCST_FILE_TYPE` and `MTD_OBS_FILE_TYPE` once those variables have been added.


Please let me know if this works for you or if you need more assistance. Thanks!

**Reply by lswenson** [reply]
Hi @georgemccabe, thanks for your response. Option 2 seems to be working well for me. Thank you so much!
